kcqrs-appengine/src/main/kotlin/com/clouway/kcqrs/adapter/appengine/AppEngineEventStore.kt

```diff
@@ -40,13 +56,22 @@ class AppEngineEventStore(private val kind: String = "Event", private val messag
 
             val snapshotKey = KeyFactory.createKey(snapshotKind, aggregateId)
 
-            val snapshot = try {
+            val snapshotEntity = try {
                 dataStore.get(transaction, snapshotKey)
             } catch (ex: EntityNotFoundException) {
                 Entity(snapshotKind, aggregateId)
             }
 
-            val aggregateIndex = snapshot.getProperty("aggregateIndex") as Long? ?: 0
+            var aggregateIndex = snapshotEntity.getProperty("aggregateIndex") as Long? ?: 0
+            val version = snapshotEntity.getProperty("version") as Long? ?: 0
+
+            if (saveOptions.createSnapshot.required && saveOptions.createSnapshot.snapshotData != null) {
+                aggregateIndex += 1
+                val snapshotData = saveOptions.createSnapshot.snapshotData
+                snapshotEntity.setUnindexedProperty("version", version + 1)
```

Comment:
The snapshot version should not be incremented unless snapshot is changed. The only object who knows about these changes is the SnapshotMapper interface.

Reply:
We are actually sure that a new snapshot was created from the check: 
```Kotlin
if (saveOptions.createSnapshot.required && saveOptions.createSnapshot.snapshotData != null)
```
I could move the version incrementation to the snapshot mapper, but it will have to receive the current version of the snapshot as well.

Reply:
I'm not able to see a reason for this as the SnapshotMapper is responsible for the creation of the snapshot and he knows what the current version is.

Reply:
Check my comment below.